### ISSUE TITLE
ModuleNotFoundError for wmi, serial.tools, pyusbcameraindex, and pynput

### ISSUE THREAD

**Issue by lzyuan168** [Nuitka/Nuitka#3274]
If you want to post a problem/bug, to help us understand and resolve your issue please check that
you have provided at least the information below, and discard up to here:

- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version

  If you are not providing the full output. The issue cannot be solved and will be considered
  `invalid`. The command outputs more than you think, and we are adding more all the time. We do not
  want to left guessing or trying things out to reproduce your issue.

```
2.4.9
Commercial: 2.6.3
Python: 3.10.15 | packaged by Anaconda, Inc. | (main, Oct  3 2024, 07:22:19) [MSC v.1929 64 bit (AMD64)]
Flavor: Anaconda Python
Executable: C:\Users\ml_luke\anaconda3\envs\boson\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 11
Version C compiler: ~\AppData\Local\Nuitka\Nuitka\Cache\DOWNLO~1\gcc\x86_64\13.2.0-16.0.6-11.0.1-msvcrt-r1\mingw64\bin\gcc.exe (gcc 13.2.0).
```

- How did you install Nuitka and Python

  Did you use pip, anaconda, deb, rpm, from source, git clone, then install into a virtualenv or
  not, this is very important usually and one thing, the above command does not tell us (yet).

```
python setup.py install from source
```

- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip list -v

```
altair==5.5.0
altgraph==0.17.4
annotated-types==0.7.0
anyio==4.6.2.post1
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==2.4.1
async-lru==2.0.4
attrs==24.2.0
babel==2.16.0
beautifulsoup4==4.12.3
bleach==6.2.0
blinker==1.9.0
cachetools==5.5.0
certifi==2024.8.30
cffi==1.17.1
charset-normalizer==3.4.0
click==8.1.7
colorama==0.4.6
comm==0.2.2
comtypes==1.4.8
contourpy==1.3.1
cv2_enumerate_cameras==1.1.17
cycler==0.12.1
debugpy==1.8.9
decorator==5.1.1
defusedxml==0.7.1
dnspython==2.7.0
email_validator==2.2.0
exceptiongroup==1.2.2
executing==2.1.0
fastapi==0.115.5
fastapi-cli==0.0.5
fastjsonschema==2.20.0
fonttools==4.55.0
fqdn==1.5.1
future==1.0.0
gitdb==4.0.11
GitPython==3.1.43
h11==0.14.0
httpcore==1.0.7
httptools==0.6.4
httpx==0.27.2
idna==3.10
ipykernel==6.29.5
ipython==8.29.0
ipywidgets==8.1.5
iso8601==2.1.0
isoduration==20.11.0
itsdangerous==2.2.0
jedi==0.19.2
Jinja2==3.1.4
json5==0.9.28
jsonpointer==3.0.0
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
jupyter==1.1.1
jupyter_client==8.6.3
jupyter-console==6.6.3
jupyter_core==5.7.2
jupyter-events==0.10.0
jupyter-lsp==2.2.5
jupyter_server==2.14.2
jupyter_server_terminals==0.5.3
jupyterlab==4.2.6
jupyterlab_pygments==0.3.0
jupyterlab_server==2.27.3
jupyterlab_widgets==3.0.13
keyboard==0.13.5
kiwisolver==1.4.7
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib==3.9.2
matplotlib-inline==0.1.7
mdurl==0.1.2
mistune==3.0.2
narwhals==1.15.1
nbclient==0.10.0
nbconvert==7.16.4
nbformat==5.10.4
nest-asyncio==1.6.0
notebook==7.2.2
notebook_shim==0.2.4
numpy==2.1.3
opencv-python==4.10.0.84
ordered-set==4.1.0
orjson==3.10.12
overrides==7.7.0
packaging==24.2
pandas==2.2.3
pandocfilters==1.5.1
parso==0.8.4
pefile==2023.2.7
pillow==11.0.0
platformdirs==4.3.6
prometheus_client==0.21.0
prompt_toolkit==3.0.48
protobuf==5.29.0
psutil==6.1.0
pure_eval==0.2.3
pyarrow==18.1.0
pycparser==2.22
pydantic==2.10.2
pydantic_core==2.27.1
pydantic-extra-types==2.10.0
pydantic-settings==2.6.1
pydeck==0.9.1
Pygments==2.18.0
pyinstaller==6.11.1
pyinstaller-hooks-contrib==2024.10
pynput==1.7.7
pyparsing==3.2.0
pyserial==3.5
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-json-logger==2.0.7
python-multipart==0.0.17
pytz==2024.2
pyusbcameraindex==1.0.1
pywin32==308
pywin32-ctypes==0.2.3
pywinpty==2.0.14
PyYAML==6.0.2
pyzmq==26.2.0
referencing==0.35.1
requests==2.32.3
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.9.4
rpds-py==0.21.0
Send2Trash==1.8.3
serial==0.0.97
setuptools==75.1.0
shellingham==1.5.4
six==1.16.0
smmap==5.0.1
sniffio==1.3.1
soupsieve==2.6
stack-data==0.6.3
starlette==0.41.3
streamlit==1.40.2
tenacity==9.0.0
terminado==0.18.1
tinycss2==1.4.0
toml==0.10.2
tomli==2.1.0
tornado==6.4.2
traitlets==5.14.3
typer==0.13.1
types-python-dateutil==2.9.0.20241003
typing_extensions==4.12.2
tzdata==2024.2
ujson==5.10.0
uri-template==1.3.0
urllib3==2.2.3
uvicorn==0.32.1
watchdog==6.0.0
watchfiles==1.0.0
wcwidth==0.2.13
webcolors==24.11.1
webencodings==0.5.1
websocket-client==1.8.0
websockets==14.1
wheel==0.44.0
widgetsnbextension==4.0.13
WMI==1.5.1
zstandard==0.23.0
```

- Many times when you get an error from Nuitka, your setup may be special

  Then even a `print("hello world")` program will not work, please try that and report that error
  instead. We do not need a report that says "Tensorflow does not work" when nothing actually works
  for you. You ought to also read the User Manual and check that your setup is actually supported
  already.

```
Once I remove wmi, serial, pyusbcameraindex, and pynput from the python script the exe file is able to run successfully, but once these packages are import in the script, the exe file will throw ModuleNotFoundError. I have tried to add various flags such as --include-module, --include-package, but none of them works.
```

- Also supply a Short, Self Contained, Correct, Example

  That demonstrates the issue i.e a small piece of code which reproduces the issue and can be run
  with out any other (or as few as possible) external dependencies. Issues without this may will
  likely be rejected without much consideration. Often this can be as simple as importing a package,
  if this is a packaging issue, try that first.

  Pointers to repositories with usage of `pipenv` are very welcome, archives with examples are too,
  e.g. if a certain package structure is needed. This must be only source code, binaries are not
  used ever by us. But beware, that e.g. we cannot click around and stuff. Still do an effort to
  make the usage obvious. Having a compile script in the repo is perfect.

  But this cannot be much code for non-commercial users, since that causes too much effort. We
  cannot just compile your code, run it and have it download a control and command center.

```
I have simplified my code to just print Hello World, without those imports I can see it printed on the command prompt, but with those imports, it will produce error.
```

- Provide in your issue the Nuitka options used

  Ideally use the `# nuitka-project:` options feature in the code, so options and example code go
  along. Alternatively state the command line.

  [Nuitka Options in the code](https://nuitka.net/doc/user-manual.html#nuitka-options-in-the-code)

```
options used:
--standalone
--follow-import
--include-data-files  (for some python helper script)
--include-module  (tried one at a time between this and --include-package)
--include-package  (tried one at a time between this and --include-module)
--report
```

- Avoid unnecessary options

  Do not use `--deployment` as it disables all potential bug catching, we are not here to replace
  what it does, remove that option the moment you face an issue and report only then.

  Do not use `--onefile` if the issue also happens with `--standalone`. Minimize the use of options
  as far as you can, please. Do not disable outputs with `--quiet` and do not disable warnings.

- Note if this is a regression

  If it used to work in an earlier version of Nuitka, please note what you know about that. Since
  git bisect is a thing for which we do a lot of error to make it usable, this will help
  dramatically to isolate the issue.

```
I did not try with other versions of nuitka
```

**Comment by kayhayen** [comment]
Can you try with the latest stable release please? 2.5.9 may already have this changed.

**Comment by lzyuan168** [comment]
Hi, I tried with 2.5.9, still the same error 

```
2.5.9
Commercial: None
Python: 3.10.15 | packaged by Anaconda, Inc. | (main, Oct  3 2024, 07:22:19) [MSC v.1929 64 bit (AMD64)]
Flavor: Anaconda Python
Executable: ~\anaconda3\envs\boson\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 11
Version C compiler: ~\AppData\Local\Nuitka\Nuitka\Cache\DOWNLO~1\gcc\x86_64\14.2.0posix-19.1.1-12.0.0-msvcrt-r2\mingw64\bin\gcc.exe (gcc 14.2.0).
```

So when I include those imports in the python script, the exe will report ModuleNotFoundError, but if I don't include those imports in the python script, the exe file will be able to print the 'hello world'

I have attached the sample python code as well as the command I use.

This is sample_main.py, which is the main file to build the exe. It will call sample_helper.py through subprocess.

```
import os
import time
import subprocess
import sys
from watchdog.observers import Observer
from watchdog.events import FileSystemEventHandler

def main():

    subprocess.Popen(['python', 'sample_helper.py'])


if __name__ == '__main__':
    main()
```

This is the helper function, which contains the problematic imports

```
import os
import time
import sys
import wmi
import json
import logging
import csv
import argparse

from serial.tools import list_ports
from pyusbcameraindex import enumerate_usb_video_devices_windows

from datetime import datetime, timezone
from pynput import keyboard


def main():
    print('Hello from sample helper')

if __name__ == '__main__':
    main()
```

Below is the command I used to compile the exe

```
python .\Nuitka-2.5.9\bin\nuitka --onefile --standalone --include-data-files=./sample_helper.py=sample_helper.py --report=report.xml sample_main.py
```

**Comment by kayhayen** [comment]
I have just executed your problematic imports fine with Nuitka 2.7rc and I believe 2.6 also does it just fine. Your executing of "python" requires for that Python to be separate of the compiled binary, and for it to contain all necessary modules. You should look at multidist, which allows you to specifiy two "main" functions, and then use subprocess on your main binary, with the helper binary name, say "smtp_helper.exe" and then you can not depend on system Python at all.

The imports you gave just worked fine for me, not sure if you ever had a real Nuitka problem to begin with, your system Python doesn't suddenly gain things. For different process forking, multiprocessing would also be a supported option. But just calling "python" is not at all usable for deployment. Most systems don't even have it.
